sdk/formrecognizer/ai-form-recognizer/test/utils/recordedClients.ts

```diff
@@ -51,66 +52,87 @@ export const testPollingOptions: PollerOptions<PollOperationState<unknown>> = {
-    },
-    // replace the SAS token
-    (recording: string): string => {
-      return recording
-        .replace(/\?sv[^"]*"/, `?sastoken"`)
-        .replace(/\?sv[^\\"]*\\"/, `?sastoken\\"`)
-        .replace(/\?sp[^"]*"/, `?sastoken"`)
-        .replace(/\?sp[^\\"]*\\"/, `?sastoken\\"`);
-    },
-  ],
-  queryParametersToSkip: [],
+export const recorderOptions: RecorderStartOptions = {
+  envSetupForPlayback,
+  sanitizerOptions: {
+    generalSanitizers: [
+      // endpoints
+      {
+        target: env["FORM_RECOGNIZER_TRAINING_CONTAINER_SAS_URL"]?.split("/")[2] || "",
+        value: envSetupForPlayback["FORM_RECOGNIZER_TRAINING_CONTAINER_SAS_URL"].split("/")[2],
+      },
+      {
+        target: env["FORM_RECOGNIZER_TESTING_CONTAINER_SAS_URL"]?.split("/")[2] || "",
+        value: envSetupForPlayback["FORM_RECOGNIZER_TESTING_CONTAINER_SAS_URL"].split("/")[2],
+      },
+      {
+        target:
+          env["FORM_RECOGNIZER_SELECTION_MARK_STORAGE_CONTAINER_SAS_URL"]?.split("/")[2] || "",
+        value:
+          envSetupForPlayback["FORM_RECOGNIZER_SELECTION_MARK_STORAGE_CONTAINER_SAS_URL"].split(
+            "/"
+          )[2],
+      },
+      // sas tokens
+      {
+        target: env["FORM_RECOGNIZER_TRAINING_CONTAINER_SAS_URL"]?.split("?")[1] || "",
+        value: envSetupForPlayback["FORM_RECOGNIZER_TRAINING_CONTAINER_SAS_URL"].split("?")[1],
+      },
+      {
+        target: env["FORM_RECOGNIZER_TESTING_CONTAINER_SAS_URL"]?.split("?")[1] || "",
+        value: envSetupForPlayback["FORM_RECOGNIZER_TESTING_CONTAINER_SAS_URL"].split("?")[1],
+      },
+      {
+        target:
+          env["FORM_RECOGNIZER_SELECTION_MARK_STORAGE_CONTAINER_SAS_URL"]?.split("?")[1] || "",
+        value:
+          envSetupForPlayback["FORM_RECOGNIZER_SELECTION_MARK_STORAGE_CONTAINER_SAS_URL"].split(
+            "?"
+          )[1],
+      },
+    ],
+  },
 };
 
-export function createRecorder(context: Context): Recorder {
-  return record(context, environmentSetup);
-}
-
 /**
  * Returns an appropriate credential depending on the value of `useAad`.
  */
 export function makeCredential(useAad: boolean): TokenCredential | AzureKeyCredential {
   return useAad
-    ? new ClientSecretCredential(env.AZURE_TENANT_ID, env.AZURE_CLIENT_ID, env.AZURE_CLIENT_SECRET)
-    : new AzureKeyCredential(env.FORM_RECOGNIZER_API_KEY);
+    ? createTestCredential()
+    : new AzureKeyCredential(assertEnvironmentVariable("FORM_RECOGNIZER_API_KEY"));
 }
 
-export function createRecordedClient<T>(
-  context: Context,
-  ctor: { new (endpoint: string, credential: TokenCredential | KeyCredential): T },
-  apiKey?: AzureKeyCredential
-): RecordedClient<T> {
-  const recorder = record(context, environmentSetup);
+export async function createRecorder(currentTest?: Test): Promise<Recorder> {
+  const recorder = new Recorder(currentTest);
+  await recorder.start(recorderOptions);
+  return recorder;
+}
+
+export async function createRecordedClient<T>(
```

Comment:
I'm glad it was simple enough to fit this into the existing `createRecordedClient` higher-order function.